packages/directlinespeech/src/createMultiBufferingPlayer.js

```diff
@@ -0,0 +1,108 @@
+// Currently, we use a triple-buffer approach.
+const NUM_BUFFER = 3;
+
+function zeroBuffer(buffer) {
+  const channels = buffer.numberOfChannels;
+
+  for (let channel = 0; channel < channels; channel++) {
+    const audioData = buffer.getChannelData(channel);
+
+    [].fill.call(audioData, 0);
+  }
+}
+
+function copyBuffer(buffer, multiChannelArrayBuffer) {
+  const channels = buffer.numberOfChannels;
+
+  for (let channel = 0; channel < channels; channel++) {
+    const arrayBuffer = multiChannelArrayBuffer[channel];
+
+    // Safari does not support AudioBuffer.copyToChannel yet.
+    if (buffer.copyToChannel) {
+      buffer.copyToChannel(arrayBuffer, channel);
+    } else {
+      const { length: arrayBufferLength } = arrayBuffer;
+      const perChannelBuffer = buffer.getChannelData(channel);
+
+      for (let offset = 0; offset < arrayBufferLength; offset++) {
+        perChannelBuffer[offset] = arrayBuffer[offset];
+      }
+    }
+  }
+}
+
+// This is a multi-buffering player. Users can keep pushing buffer to us.
+// We will realize the buffer as BufferSource and queue it to AudioContext.
+// We will queue as soon, and as much as possible.
+// We do not support progressive buffering (push partial buffer) and do not have plan for it.
```

Comment:
Can we talk in person to clarify this one? I want to ask you some questions before giving phrasing suggestions.